test/regress_ws.c

```diff
@@ -388,4 +390,5 @@ http_ws_test(void *arg)
 end:
 	if (bev)
 		bufferevent_free(bev);
+	EVUTIL_CLOSESOCKET(fd);
```

Comment:
`if (fd != EVUTIL_INVALID_SOCKET)` (and use it for initilzation), and also let's add a check for `http`

Reply:
I noticed that the `http` is not used here? I'm not quite sure where I'm supposed to add the check. 
Additionally, I should initialize fd to `EVUTIL_INVALID_SOCKET` and check if it is `EVUTIL_INVALID_SOCKET` before closing,  is that right?

Reply:
> I noticed that the http is not used here?

It is used it creates http server, see usage of `port`.

> I'm not quite sure where I'm supposed to add the check.

The `http` initialization may fail, so - `if (!http) goto err`

>Additionally, I should initialize fd to EVUTIL_INVALID_SOCKET and check if it is EVUTIL_INVALID_SOCKET before closing, is that right?

Yes